ecosystem/indexer/src/default_processor.rs

```diff
@@ -0,0 +1,142 @@
+    schema,
+};
+
+use aptos_rest_client::Transaction;
+use async_trait::async_trait;
+use diesel::Connection;
+use futures::future::Either;
+use std::{fmt::Debug, sync::Arc};
+
+pub struct DefaultTransactionProcessor {
+    connection_pool: PgDbPool,
+}
+
+impl DefaultTransactionProcessor {
+    pub fn new(connection_pool: PgDbPool) -> Self {
+        Self { connection_pool }
+    }
+}
+
+impl Debug for DefaultTransactionProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "DefaultTransactionProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+#[async_trait]
+impl TransactionProcessor for DefaultTransactionProcessor {
+    fn name(&self) -> &'static str {
+        "default_processor"
+    }
+
+    async fn process_transaction(
+        &self,
+        transaction: Arc<Transaction>,
+    ) -> Result<ProcessingResult, TransactionProcessingError> {
+        let version = transaction.version().unwrap_or(0);
+
+        let (transaction_model, maybe_details_model, maybe_events) =
+            TransactionModel::from_transaction(&transaction);
+
+        let conn = self.connection_pool.get().map_err(|e| {
+            TransactionProcessingError::ConnectionPoolError((
+                anyhow::Error::from(e),
+                version,
+                self.name(),
+            ))
+        })?;
+
+        let tx_result = conn.transaction::<(), diesel::result::Error, _>(||{
+        aptos_logger::trace!(
+            "[default_processor] inserting 'transaction' version {} with hash {}",
+            version,
+            &transaction_model.hash
+        );
+        execute_with_better_error(
+            &conn,
+            diesel::insert_into(schema::transactions::table)
+                .values(&transaction_model)
+                .on_conflict(schema::transactions::dsl::hash)
+                .do_update()
+                .set(&transaction_model),
+        )
+        .expect("Error inserting row into database");
+
+        if let Some(tx_details_model) = maybe_details_model {
+            match tx_details_model {
+                Either::Left(ut) => {
+                    aptos_logger::trace!(
+                        "[default_processor] inserting 'user_transaction' version {} with hash {}",
+                        version,
+                        &transaction_model.hash
+                    );
+                    execute_with_better_error(
+                        &conn,
+                        diesel::insert_into(schema::user_transactions::table)
+                            .values(&ut)
+                            .on_conflict(schema::user_transactions::dsl::hash)
+                            .do_update()
+                            .set(&ut),
+                    )
+                    .expect("Error inserting row into database");
```

Comment:
nit: you could always put the expect into `execute_with_better_error` to avoid repeating it everywhere 😄 The panic should provide a backtrace, so you could figure out where it died.

Reply:
I wanted to keep it flexible despite using `.expect` everywhere haha. In a future PR I can see removing all of these expects and things returning results :-)